apps/metadata-editor/src/app/records/my-org-records/my-org-records.component.spec.ts

```diff
@@ -1,108 +1,164 @@
+    },
+  ],
+}
 
-class AuthServiceMock {
-  user$ = new BehaviorSubject(user)
-  authReady = jest.fn(() => this._authSubject$)
-  _authSubject$ = new BehaviorSubject({})
+const myOrgServiceMock = {
+  myOrgData$: of(orgDataMock),
 }
-class OrganisationsServiceMock {
-  getFiltersForOrgs = jest.fn(() => new BehaviorSubject(filters))
-  organisationsCount$ = of(456)
+
+const user = USER_FIXTURE_ANON()
+const allUsers = USERS_FIXTURE()
+
+const authServiceMock = {
+  user$: new BehaviorSubject(user),
+  allUsers$: new BehaviorSubject(allUsers),
 }
 
-class searchServiceMock {
-  updateSearchFilters = jest.fn()
-  setSearch = jest.fn()
-  setSortBy = jest.fn()
-  setSortAndFilters = jest.fn()
+const organisationsServiceMock = {
+  organisations$: of(ORGANISATIONS_FIXTURE),
 }
 
-class SearchFacadeMock {
-  resetSearch = jest.fn()
-  setFilters = jest.fn()
+const searchFacadeMock = {
+  resetSearch: jest.fn(),
 }
 
-@Component({
-  // eslint-disable-next-line
-  selector: 'md-editor-records-list',
-  template: '',
-  standalone: true,
-})
-export class MockRecordsListComponent {}
+const routeServiceMock = {
+  getDatahubSearchRoute: jest.fn(),
+}
 
 describe('MyOrgRecordsComponent', () => {
   let component: MyOrgRecordsComponent
-  let fixture: ComponentFixture<MyOrgRecordsComponent>
   let searchFacade: SearchFacade
-  let orgService: OrganizationsServiceInterface
+  let myOrgService: MyOrgService
+  let authService: AuthService
+  let orgServiceInterface: OrganizationsServiceInterface
+  let routerService: EditorRouterService
 
   beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [
-        importProvidersFrom(TranslateModule.forRoot()),
-        {
-          provide: SearchFacade,
-          useClass: SearchFacadeMock,
-        },
-        { provide: AuthService, useClass: AuthServiceMock },
-        {
-          provide: OrganizationsServiceInterface,
-          useClass: OrganisationsServiceMock,
-        },
-        {
-          provide: SearchFacade,
-          useClass: SearchFacadeMock,
-        },
-        {
-          provide: SearchService,
-          useClass: searchServiceMock,
-        },
-      ],
-    }).overrideComponent(MyOrgRecordsComponent, {
-      remove: {
-        imports: [RecordsListComponent],
-      },
-      add: {
-        imports: [MockRecordsListComponent],
-      },
-    })
-    searchFacade = TestBed.inject(SearchFacade)
-    orgService = TestBed.inject(OrganizationsServiceInterface)
-    fixture = TestBed.createComponent(MyOrgRecordsComponent)
-    component = fixture.componentInstance
-    fixture.detectChanges()
+    orgServiceInterface = organisationsServiceMock as any
+    myOrgService = myOrgServiceMock as any
+    authService = authServiceMock as any
+    searchFacade = searchFacadeMock as any
+    routerService = routeServiceMock as any
+
+    component = new MyOrgRecordsComponent(
+      myOrgService,
+      searchFacade,
+      orgServiceInterface,
+      routerService
+    )
```

Comment:
I was wondering what was a reason here for not using the `TestBed` class?

Reply:
Oh, I find it easier and lighter without a `TestBed` actually. Since these tests don't require interactions with the DOM (as all UT should in fact, since we have integration tests) and it was a brand new test file I thought of giving it a goo :slightly_smiling_face:  But I can change it to align with the rest of the app.